### DISCUSSION TITLE
Adding shutdown as a button action

### DISCUSSION THREAD

**Discussion by demusss** [sethvoltz/OctoPrint-DisplayPanel#78] [Ideas]
I'm using a Pi Zero 2 W with an Adafruit 128x64 oled bonnet as the screen. The bonnet has 7 switches in total.

Is it possible to add a button action to shutdown OctoPrint? Similar to how you've done the cancel print where you have to confirm within 10 seconds.

It would be useful as the only way of cleanly shutting down OctoPrint is to load up a PC or login via mobile. I don't like just pulling the plug.

Congratulations on the new born btw :)

**Comment by sethvoltz** [comment]
Great idea! The controller plug-in is currently tied to the idea of 4 buttons so it would take some doing to clean it up further and add arbitrary buttons to the base controller, virtual and physical classes. That said, if you wanted to take a crack at it, I'm happy to review any pulls. 

In the meantime, you might be able to get what you want from this plug-in: https://plugins.octoprint.org/plugins/physicalbutton/ -- while I haven't tested this one myself, other plug-ins that access the GPIO should work fine alongside mine and that way you can have the main controls my UI offers as well as configure the additional buttons for whatever you need.

**Comment by Hairyloon** [comment]
I haven't looked at the code, but I would have thought you should be able to have a reboot option by using the buttons in combination.
Perhaps hold three buttons together, then we're following the tradition of having the three finger salute.
